src/target/stm32l4.c

```diff
@@ -124,6 +124,108 @@ struct stm32l4_flash {
 	uint32_t bank1_start;
 };
 
+enum ID_STM32L4 {
+	ID_STM32L41  = 0x464u, /* RM0394, Rev.4 */
+	ID_STM32L43  = 0x435u, /* RM0394, Rev.4 */
+	ID_STM32L45  = 0x462u, /* RM0394, Rev.4 */
+	ID_STM32L47  = 0x415u, /* RM0351, Rev.5 */
+	ID_STM32L49  = 0x461u, /* RM0351, Rev.5 */
+	ID_STM32L4R  = 0x470u, /* RM0432, Rev.5 */
+	ID_STM32G07  = 0x460u, /* RM0444/454, Rev.1 */
+};
+
+enum FAM_STM32L4 {
+	FAM_STM32L4xx = 1,
+	FAM_STM32L4Rx = 2,
+	FAM_STM32G0x = 3,
+	FAM_STM32WBxx = 4,
+};
+
+#define DUAL_BANK	0x80u
+#define RAM_COUNT_MSK	0x07u
+
+struct stm32l4_info {
+	char *designator;
```

Comment:
The designator has constant length. In my understanding, char designator[10] will save us a pointer for every entry. Will fill up flash fast...

Reply:
Yes, very good observation.
I don't think there'll be a designator longer than 9 char, so I agree.
More or less 3 bytes per entry saved, no change to the code.

Reply:
Note that my solution was already less expensive than the original (+data -code)!
```
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       2019-03-03     14:42          72284 blackmagic.bin <- master branch
-a----       2019-03-03     14:40          72192 blackmagic.nb  <- newbrain, char*
-a----       2019-03-03     14:38          72228 blackmagic.old <- newbrain char[10]
```